### PR TITLE
Update to CMSIS-RTOS2 API v2.3.0

### PR DESCRIPTION
**Pull request by RobertRostohar**
- processor affinity not supported

### CODE REVIEW

File: Documentation/Doxygen/src/history.md [hunk 1]
```diff
@@ -10,24 +10,19 @@ The table below provides information about the changes delivered with specific v
...
        - Added support for Process Isolation: MPU Protected Zones, Safety Classes, Thread Watchdogs.
        - Requires CMSIS:OS Tick:SysTick component.
        - Reduced component variants: Library (Library_NS replacement), Source (Source_NS replacement).
        - Libraries for Arm Compiler 6 built with default wchar and enums size.
        - Removed support for legacy Arm Compiler 5.
        - Fixed osMessageQueueGetSpace when called from ISR after osMessageQueuePut and before returning from ISR.
-       </td>
-    </tr>
-    <tr>
-      <td>V5.5.5</td>
```
Comment (JonatanAntoni) [inline Documentation/Doxygen/src/history.md:25]: v5.5.5 was never released

File: Documentation/Doxygen/src/history.md [hunk 2]
```diff
@@ -10,24 +10,19 @@ The table below provides information about the changes delivered with specific v
       <th>Description</th>
     </tr>
     <tr>
-      <td>V5.7.0</td>
```
Comment (JonatanAntoni) [inline Documentation/Doxygen/src/history.md:13]: v5.7.0 was never released

File: Documentation/Doxygen/src/history.md (hunk 1 above)
Comment (RobertRostohar) [reply Documentation/Doxygen/src/history.md:25]: Therefore it is removed with this commit.

File: Documentation/Doxygen/src/history.md (hunk 2 above)
Comment (RobertRostohar) [reply Documentation/Doxygen/src/history.md:13]: Therefore it is removed with this commit.

**Review by VladimirUmek — APPROVED** [review]
Looks fine.

**Pull request merged**